code/modules/clothing/glasses/glasses.dm

```diff
@@ -291,6 +291,17 @@
 	flags = NODROP
 	invis_view = SEE_INVISIBLE_MINIMUM
 
+/obj/item/clothing/glasses/godeye/attackby(obj/item/weapon/W as obj, mob/user as mob, params)
+	if(istype(W, src) && W != src && W.loc == user)
+		if(W.icon_state == "godeye")

```

Comment:
It's behind an istype check that limits it to `/godeye` s. Also you're implying that "godeye" is a icon_state that makes sense in any other capacity.

It saves an object var man, an object var!